readthedocs/projects/models.py

```diff
@@ -570,8 +572,15 @@ def has_aliases(self):
         return self.aliases.exists()
 
     def has_pdf(self, version_slug=LATEST):
+        version = self.versions.filter(slug=version_slug).first()
         if not self.enable_pdf_build:
```

Comment:
We can probably just remove the initial check on this, and simply look for the file existing. I believe the issue is that we'd have PDF's sitting on disk that were stale, so we added this flag. Another option is to set this flag in the database based off the YAML file.